rust/toolchain.bzl

```diff
@@ -17,6 +27,42 @@ def _rust_toolchain_impl(ctx):
     for k, v in ctx.attr.debug_info.items():
         if not k in ctx.attr.opt_level:
             fail("Compilation mode {} is not defined in opt_level but is defined debug_info".format(k))
+    link_inputs = []
```

Comment:
I'd extract this whole logic into a function retuning CcInfo or None and call it in the provider constructor.